libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -25,14 +26,19 @@ const PRISMA_FORMAT_HINT: &str = "You can run `prisma format` to fix this automa
 
 impl<'a> Validator<'a> {
     /// Creates a new instance, with all builtin attributes registered.
-    pub fn new(source: Option<&'a configuration::Datasource>) -> Validator<'a> {
+    pub(crate) fn new(source: Option<&'a configuration::Datasource>) -> Validator<'a> {
         Self { source }
     }
 
-    pub fn validate(&self, ast_schema: &ast::SchemaAst, schema: &mut dml::Datamodel) -> Result<(), Diagnostics> {
+    pub(crate) fn validate(
+        &self,
+        ast_schema: &ast::SchemaAst,
+        names: &Names<'_>,
```

Comment:
Keeping the changes to a reviewable amount for this PR, but ultimately, the AST and components like `Names` should be grouped into a struct for convenience. It could be called something like `ValidationContext<'ast>`.